File: e2.cs
using System;

public class e2
{
	public static int fun(ref int k) 
	{
		k += 4;
		return 3 * k - 1;
	}

	public static void Main()
	{
		int i = 10, j = 10, sum1, sum2;
		sum1 = (i / 2) + fun(ref i); //46
		sum2 = fun(ref j) + (j / 2); //48

		Console.WriteLine("sum1: "+sum1);
		Console.WriteLine("sum2: "+sum2);
	}
}

File: e2.java
import java.util.*;
import java.lang.*;
import java.io.*;

class e2
{
	public static int fun(ref k) 
	{
		k.var += 4;
		return 3 * k.var - 1;
	}

	public static void main (String[] args)
	{
		ref i=new ref(10); 
		ref j=new ref(10); 
		int sum1, sum2;
		sum1 = (i.var / 2) + fun(i);
		sum2 = fun(j) + (j.var / 2);

		System.out.println("sum1: "+sum1);
		System.out.println("sum2: "+sum2);
	}
}